boa_tester/src/edition.rs

```diff
@@ -0,0 +1,332 @@
+    "String.prototype.at" => SpecEdition::ES13,
+    "String.prototype.endsWith" => SpecEdition::ES6,
+    "String.prototype.includes" => SpecEdition::ES6,
+    "String.prototype.matchAll" => SpecEdition::ES13,
+    "String.prototype.replaceAll" => SpecEdition::ES13,
+    "String.prototype.trimEnd" => SpecEdition::ES13,
+    "String.prototype.trimStart" => SpecEdition::ES13,
+    "super" => SpecEdition::ES6,
+    "Symbol" => SpecEdition::ES6,
+    "Symbol.asyncIterator" => SpecEdition::ES13,
+    "Symbol.hasInstance" => SpecEdition::ES6,
+    "Symbol.isConcatSpreadable" => SpecEdition::ES6,
+    "Symbol.iterator" => SpecEdition::ES6,
+    "Symbol.match" => SpecEdition::ES6,
+    "Symbol.matchAll" => SpecEdition::ES13,
+    "Symbol.prototype.description" => SpecEdition::ES13,
+    "Symbol.replace" => SpecEdition::ES6,
+    "Symbol.search" => SpecEdition::ES6,
+    "Symbol.species" => SpecEdition::ES6,
+    "Symbol.split" => SpecEdition::ES6,
+    "Symbol.toPrimitive" => SpecEdition::ES6,
+    "Symbol.toStringTag" => SpecEdition::ES6,
+    "Symbol.unscopables" => SpecEdition::ES6,
+    "tail-call-optimization" => SpecEdition::ES6,
+    "template" => SpecEdition::ES6,
+    "top-level-await" => SpecEdition::ES13,
+    "TypedArray" => SpecEdition::ES6,
+    "TypedArray.prototype.at" => SpecEdition::ES13,
+    "u180e" => SpecEdition::ES5,
+    "Uint8Array" => SpecEdition::ES6,
+    "Uint16Array" => SpecEdition::ES6,
+    "Uint32Array" => SpecEdition::ES6,
+    "Uint8ClampedArray" => SpecEdition::ES6,
+    "WeakMap" => SpecEdition::ES6,
+    "WeakRef" => SpecEdition::ES13,
+    "WeakSet" => SpecEdition::ES6,
+    "well-formed-json-stringify" => SpecEdition::ES5,
+    "__proto__" => SpecEdition::ES6,
+    "__getter__" => SpecEdition::ES13,
+    "__setter__" => SpecEdition::ES13,
+
+    // Test-Harness Features
+
+    "IsHTMLDDA" => SpecEdition::ES13,
+    "host-gc-required" => SpecEdition::ES13,
+};
+
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Default,
+    Serialize,
+    Deserialize,
+    clap::ValueEnum,
+)]
+#[serde(untagged)]
+pub(crate) enum SpecEdition {
+    ES5 = 5,
+    ES6,
+    ES7,
+    ES8,
+    ES9,
+    ES10,
+    ES11,
+    ES12,
+    #[default]
+    ES13,
+}
+
+impl Display for SpecEdition {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match *self {
+            Self::ES5 => "ECMAScript 5",
+            Self::ES6 => "ECMAScript 6",
+            Self::ES7 => "ECMAScript 7",
+            Self::ES8 => "ECMAScript 8",
+            Self::ES9 => "ECMAScript 9",
+            Self::ES10 => "ECMAScript 10",
+            Self::ES11 => "ECMAScript 11",
+            Self::ES12 => "ECMAScript 12",
+            Self::ES13 => "ECMAScript 13",
+        }
+        .fmt(f)
+    }
+}
+
+impl SpecEdition {
+    pub(crate) fn from_test_metadata(metadata: &MetaData) -> Self {
+        if metadata.es5id.is_some() {
```

Comment:
This is usually true, but it might happen that a test has multiple features, and some of them are ES5, while others are ES6, not sure if we're taking that into account and report the higher version among the features in the test.